### PR TITLE
vrouter: remove a POSTROUTING rule for port forwarding in VPC router

### PR DESCRIPTION
**Pull request by ustcweizhou**
## Description

As discussed in https://github.com/apache/cloudstack/pull/3937#issuecomment-595459008
a rule for port forwarding in VPC router might not be needed.

This fixes the failed result of health check for network VRs.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by weizhouapache** [comment]
@DaanHoogland can you kick off test for this PR ?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: systemvm/debian/opt/cloud/bin/configure.py [hunk 1]
```diff
@@ -918,15 +918,6 @@ def forward_vpc(self, rule):
         if not rule["internal_ports"] == "any":
             fw_prerout_rule += ":" + self.portsToString(rule["internal_ports"], "-")
 
-        fw_postrout_rule = "-A POSTROUTING -d %s/32 " % rule["public_ip"]
```
Comment (DaanHoogland) [inline systemvm/debian/opt/cloud/bin/configure.py]: pylint error: no longer defined but still used in 942

Comment (weizhouapache) [reply systemvm/debian/opt/cloud/bin/configure.py]: should be fixed now

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by andrijapanicsb** [comment]
Testing this one, instead of #3937 - we removed the offending rule, so additional code to handle VPC vs Isolated network VR is not needed.

**Comment by andrijapanicsb** [comment]
@weizhouapache testing this one - and after configuring port forward (and stopping iptables on the guest VM) - I can't connect via PF rule to the VM (works fine for VPC though).

Can you please take a look? Otherwise, since we are 1 day from the freeze for 4.14., I'll have to go with #3937 (but I prefer your PR with removed extra lines) ?

**Comment by weizhouapache** [comment]
> @weizhouapache testing this one - and after configuring port forward (and stopping iptables on the guest VM) - I can't connect via PF rule to the VM (works fine for VPC though).
> 
> Can you please take a look? Otherwise, since we are 1 day from the freeze for 4.14., I'll have to go with #3937 (but I prefer your PR with removed extra lines) ?

@andrijapanicsb I will look into it today.
one suggestion, can we easily remove a line in systemvm/debian/root/health_checks/iptables_check.py as a workaround ? health check will pass in both network and vpc vr.

https://github.com/apache/cloudstack/pull/3952/files#diff-cf7b342df9a0681cd60e083d7f746bd2

**Comment by andrijapanicsb** [comment]
I see it should work @weizhouapache , but let me ping @Pearl1594 for the opinion

thx

**Comment by andrijapanicsb** [comment]
We'll do another PR with just that one change, thanks @weizhouapache

**Comment by andrijapanicsb** [comment]
@weizhouapache the PR was made, tested (by you and @Pearl1594 - thanks ) and merged - here  https://github.com/apache/cloudstack/pull/3963 

Do you want to close this one now or... ?
I will remove the milestone from it - thanks!

**Comment by weizhouapache** [comment]
> @weizhouapache the PR was made, tested (by you and @Pearl1594 - thanks ) and merged - here #3963
> 
> Do you want to close this one now or... ?
> I will remove the milestone from it - thanks!

considering we have merged #3963 , I think we can move this to 4.14.1

by the way, I have tested the failed test test_vpc_redundant.py in my local testing environment, no issue at all.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@ustcweizhou can you review the test failures?

**Comment by weizhouapache** [comment]
> @ustcweizhou can you review the test failures?

@rhtyd can you re-kick off the tests ?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Test LGTM the failure is due to a regression in Trillian/ansible overiding file (the test_vm_life_cycle.py and has been fixed).

**Comment by rhtyd** [comment]
The regression has been fixed, failures not related to this PR.

**Pull request merged**